iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/task/PipeDataNodeTaskAgent.java

```diff
@@ -111,6 +111,54 @@ protected Map<Integer, PipeTask> buildPipeTasks(final PipeMeta pipeMetaFromConfi
     return new PipeDataNodeBuilder(pipeMetaFromConfigNode).build();
   }
 
+  ////////////////////////// Manage by Pipe Name //////////////////////////
+
+  @Override
+  protected void startPipe(final String pipeName, final long creationTime) {
+    final PipeMeta existedPipeMeta = pipeMetaKeeper.getPipeMeta(pipeName);
+    final PipeStatus status = existedPipeMeta.getRuntimeMeta().getStatus().get();
+    if (PipeStatus.STOPPED.equals(status) || status == null) {
+      restartPipeToReloadResourceIfNeeded(existedPipeMeta);
+    }
+
+    super.startPipe(pipeName, creationTime);
+  }
+
+  private void restartPipeToReloadResourceIfNeeded(final PipeMeta pipeMeta) {
+    if (System.currentTimeMillis() - pipeMeta.getStaticMeta().getCreationTime()
+        < PipeConfig.getInstance().getPipeStuckRestartMinIntervalMs()) {
+      return;
+    }
+
+    final AtomicLong lastRestartTime =
+        PIPE_NAME_TO_LAST_RESTART_TIME_MAP.get(pipeMeta.getStaticMeta().getPipeName());
+    if (lastRestartTime != null
+        && System.currentTimeMillis() - lastRestartTime.get()
+            < PipeConfig.getInstance().getPipeStuckRestartMinIntervalMs()) {
+      LOGGER.info(
+          "Skipping reload resource for stopped pipe {} before starting it because reloading resource is too frequent.",
+          pipeMeta.getStaticMeta().getPipeName());
+      return;
+    }
+
+    if (PIPE_NAME_TO_LAST_RESTART_TIME_MAP.isEmpty()) {
+      LOGGER.info(
+          "Flushing storage engine before restarting pipe {}.",
+          pipeMeta.getStaticMeta().getPipeName());
+      final long currentTime = System.currentTimeMillis();
+      StorageEngine.getInstance().syncCloseAllProcessor();
+      WALManager.getInstance().syncDeleteOutdatedFilesInWALNodes();
```

Comment:
Seemingly this is for dynamic flushing? Will this cause more small tsFiles?(i.e. flush here, and when collecting historical tsFiles)